controlsfx/src/main/java/org/controlsfx/control/textfield/TextFields.java

```diff
@@ -109,6 +110,10 @@ private static void setupClearButtonField(TextField inputField, ObjectProperty<N
 
         inputField.textProperty().addListener(new InvalidationListener() {
             @Override public void invalidated(Observable arg0) {
+                if (fader.getStatus() == Animation.Status.RUNNING) {
+                    fader.stop();
+                    fader.getNode().setOpacity(fader.getToValue());
+                }

```

Comment:
Why checking the fader state in the first place?
You could store the button's visible state in a boolean field and use that to switch the state and update the fader.

Reply:
Hi @goldfischli

Can you add a comment with the suggested code changes ?

Reply:
I was thinking about something like this:
```java
        inputField.textProperty().addListener(new InvalidationListener() {
            private boolean isButtonVisible = false;

            @Override public void invalidated(Observable arg0) {
                String text = inputField.getText();
                boolean isTextEmpty = text == null || text.isEmpty();

                if (isTextEmpty == isButtonVisible)
                {
                    isButtonVisible = !isTextEmpty;
                    fadeTo(isButtonVisible);
                }
            }

            private void fadeTo( boolean visible ) {
                fader.stop();
                fader.setFromValue(visible? 0.0: 1.0);
                fader.setToValue(visible? 1.0: 0.0);
                fader.play();
            }
        });
```
What do you think about it?

Reply:
A good solution indeed. I will update the PR.